sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/aggregate/PercentileSuite.scala

```diff
@@ -232,6 +358,41 @@ class PercentileSuite extends SparkFunSuite {
     assert(agg.eval(buffer) != null)
   }
 
+  test("null and invalid values( 0 and negatives ) handling of frequency column") {
```

Comment:
Just make this test about testing the negative frequency case & move null & 0 testing to the earlier test. This should be enough:
```scala
val perc = new Percentile(Literal(1), Literal(0.5), Literal(-1))
val buffer = new GenericInternalRow(2)
agg.initialize(buffer)
val e = intercept[SparkException](agg.update(buffer, InternalRow.empty))
assert(e.getMessage.contains("Negative values found in "))
```